lib/ex_unit/lib/ex_unit/cli_formatter.ex

```diff
@@ -210,14 +210,22 @@ defmodule ExUnit.CLIFormatter do
     do: colorize([:bright, :black], msg, config)
 
   defp formatter(:diff_delete, msg, config),
-    do: colorize(:red, msg, config)
+    do: colorize(:red, show_whitespace(msg), config)
 
   defp formatter(:diff_insert, msg, config),
-    do: colorize(:green, msg, config)
+    do: colorize(:green, show_whitespace(msg), config)
 
   defp formatter(_,  msg, _config),
     do: msg
 
+  defp show_whitespace(msg) do
+    if String.match?(msg, ~r/\A[ ]+\z/) do
```

Comment:
If we run the replacement on any fragment rather than just the ones that are entirely whitespace we end up with diffs like this:

![2016-05-26-100500_479x164_scrot](https://cloud.githubusercontent.com/assets/6134406/15569542/42b02ec8-2329-11e6-87b1-daa7f306e580.png)

See how the whitespace in this non-string data structure has been highlighted.

Is there a better way to do this? I wanted to run a regex substitution that would do both the check and replacement in one call, but I was unsure how I could insert the correct number of characters is this fashion.